core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java

```diff
@@ -196,56 +193,66 @@ public void testRollbacks() {
     filesMatch(Lists.newArrayList("B", "D", "E"), appendsBetweenScan(1, 5));
     filesMatch(Lists.newArrayList("B", "D", "E"), appendsAfterScan(1));
   }
 
-  @Test
+  @TestTemplate
   public void testIgnoreResiduals() throws IOException {
     add(table.newAppend(), files("A"));
     add(table.newAppend(), files("B"));
     add(table.newAppend(), files("C"));
 
-    TableScan scan1 = table.newScan().filter(Expressions.equal("id", 5)).appendsBetween(1, 3);
+    IncrementalAppendScan scan1 =
+        table
+            .newIncrementalAppendScan()
+            .filter(Expressions.equal("id", 5))
+            .fromSnapshotExclusive(1)
+            .toSnapshot(3);
 
     try (CloseableIterable<CombinedScanTask> tasks = scan1.planTasks()) {
-      Assert.assertTrue(
-          "Tasks should not be empty", com.google.common.collect.Iterables.size(tasks) > 0);
+      assertThat(tasks).as("Tasks should not be empty").hasSizeGreaterThan(0);
       for (CombinedScanTask combinedScanTask : tasks) {
         for (FileScanTask fileScanTask : combinedScanTask.files()) {
-          Assert.assertNotEquals(
-              "Residuals must be preserved", Expressions.alwaysTrue(), fileScanTask.residual());
+          assertThat(fileScanTask.residual())
+              .as("Residuals must be preserved")
+              .isNotEqualTo(Expressions.alwaysTrue());
         }
       }
     }
 
-    TableScan scan2 =
-        table.newScan().filter(Expressions.equal("id", 5)).appendsBetween(1, 3).ignoreResiduals();
+    IncrementalAppendScan scan2 =
+        table
+            .newIncrementalAppendScan()
+            .filter(Expressions.equal("id", 5))
+            .fromSnapshotExclusive(1)
+            .toSnapshot(3)
+            .ignoreResiduals();
 
     try (CloseableIterable<CombinedScanTask> tasks = scan2.planTasks()) {
-      Assert.assertTrue(
-          "Tasks should not be empty", com.google.common.collect.Iterables.size(tasks) > 0);
+      assertThat(tasks).as("Tasks should not be empty").hasSizeGreaterThan(0);
       for (CombinedScanTask combinedScanTask : tasks) {
         for (FileScanTask fileScanTask : combinedScanTask.files()) {
-          Assert.assertEquals(
-              "Residuals must be ignored", Expressions.alwaysTrue(), fileScanTask.residual());
+          assertThat(fileScanTask.residual())
+              .as("Residuals must be ignored")
+              .isEqualTo(Expressions.alwaysTrue());
         }
       }
     }
   }
 
-  @Test
-  public void testPlanWithExecutor() throws IOException {
+  @TestTemplate
+  public void testPlanWithExecutor() {
     add(table.newAppend(), files("A"));
     add(table.newAppend(), files("B"));
     add(table.newAppend(), files("C"));
 
     AtomicInteger planThreadsIndex = new AtomicInteger(0);
-    TableScan scan =
+    IncrementalAppendScan scan =
         table
-            .newScan()
-            .appendsAfter(1)
+            .newIncrementalAppendScan()
+            .fromSnapshotExclusive(1)
```

Comment:
why was this changed?

Reply:
As I commented above, `table.newScan().appendsBetween` is deprecated. Revert this part to the previous one.